Possible regression for Nix packages

I think the latest changes might have introduced a regression regarding builds with the Nix package manager. I have not had a chance to look at builds in Fedora. I currently don't have any specifics, just symptoms.

Typical problem is, that the packages can't be picked up (Note: this includes printing the flags I changed myself):
```
% result/bin/purebred                                                            
Configuration '/home/rjoost/.config/purebred/purebred.hs' changed. Recompiling.
Flags: -v0-i/home/rjoost/.config/purebred/lib-i/home/rjoost/works/purebred-i/nix/store/y5g5s6q926lsj6h9dyjgpdq91h3bwhzr-purebred-0.1.0.0/lib/ghc-8.8.4/x86_64-linux-ghc-8.8.4/purebred-0.1.0.0-7Yxwc0odFxzINrVJ1QtwAR-threaded--make/home/rjoost/.config/purebred/purebred.hs-ou
tputdir/home/rjoost/.cache/purebred-o/home/rjoost/.cache/purebred/purebred-linux-x86_64.tmp-fforce-recomp
Error occurred while loading configuration file.
purebred: 
/home/rjoost/.config/purebred/purebred.hs:24:1: error:
    Could not find module ‘Purebred’
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
24 | import Purebred
   | ^^^^^^^^^^^^^^^

/home/rjoost/.config/purebred/purebred.hs:25:1: error:
    Could not find module ‘Purebred.Plugin.ICU’
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
25 | import qualified Purebred.Plugin.ICU
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/rjoost/.config/purebred/purebred.hs:33:1: error:
    Could not find module ‘Data.MIME’
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
33 | import Data.MIME (matchContentType)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

CallStack (from HasCallStack):
  error, called at src/Purebred.hs:347:69 in purebred-0.1.0.0-7Yxwc0odFxzINrVJ1QtwAR:Purebred
```

--- a/Config/Dyre/Compile.hs
+++ b/Config/Dyre/Compile.hs
@@ -4,6 +4,7 @@ deals with error handling, and not the compilation itself /per se/.
 -}
 module Config.Dyre.Compile ( customCompile, getErrorPath, getErrorString ) where
 
+import Control.Applicative ((<|>))
 import Control.Concurrent ( rtsSupportsBoundThreads )
 import Control.Monad (when)
 import Data.Maybe (fromMaybe)
@@ -64,7 +65,9 @@ customCompile params@Params{statusOut = output} = do
             then return $ Just stackYamlPath
             else return Nothing
 
-        hc <- fromMaybe "ghc" <$> lookupEnv "HC"
+        hc' <- lookupEnv "HC"
+        nix_ghc <- lookupEnv "NIX_GHC"
+        let hc = fromMaybe "ghc" (hc' <|> nix_ghc)
         ghcProc <- maybe (runProcess hc flags (Just cacheDir') Nothing
                               Nothing Nothing (Just errHandle))
                          (\stackYaml' -> runProcess "stack" ("ghc" : "--stack-yaml" : stackYaml' : "--" : flags)

